web-client/src/presenter/actions/PDFPreviewTab/loadPdfForTabAction.ts

```diff
@@ -0,0 +1,44 @@
+export const loadPdfForTabAction = ({
+  applicationContext,
+  props,
+  router,
+}: ActionProps) => {
+  const { file } = props;
+  const isBase64Encoded = typeof file === 'string' && file.startsWith('data');
+
+  return new Promise<void>((resolve, reject) => {
+    const reader = applicationContext.getFileReaderInstance();
+
+    reader.onload = () => {
+      let binaryFile: string | ArrayBuffer | null;
+      if (isBase64Encoded && reader.result === 'string') {
+        const base64File = reader.result.replace(/[^,]+,/, '');
+        binaryFile = atob(base64File);
+      } else {
+        binaryFile = reader.result;
+      }
+
+      try {
+        const pdfDataUri = router.createObjectURL(
+          // @ts-ignore
+          new Blob([binaryFile], { type: 'application/pdf' }),
+        );
+        window.open(pdfDataUri, '_blank');
+        resolve();
+      } catch (err) {
+        reject(err);
+      }
+    };
+
+    reader.onerror = function (err) {
+      reject(err);
+    };
+
+    if (isBase64Encoded) {
+      // @ts-ignore
+      reader.readAsDataURL(file);
+    } else {
+      reader.readAsArrayBuffer(file);
+    }
```

Comment:
I would spend some time investigating this since the type mismatch looks like a mistake. `readAsDataURL` expects a `Blob` or `file` ... but file is a `string` here. It might be as simple as something like `reader.readAsText(new Blob([file]));`, but I'm not sure.

Reply:
Checked on this. Awaiting for test env to merge